{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![flybase](http://gannet.fish.washington.edu/seashell/snaps/FlyBase_Homepage_2021-01-30_08-23-39.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> ../blastdb/dmel-all-gene-r6.37.fasta <==\n",
      ">FBgn0031081 type=gene; loc=X:19961297..19969323; ID=FBgn0031081; name=Nep3; dbxref=FlyBase:FBan0009565,FlyBase:FBgn0031081,FlyBase_Annotation_IDs:CG9565,GB_protein:AAF45370,INTERPRO:IPR018497,GB:AY095202,GB_protein:AAM12295,GB:BI363270,UniProt/Swiss-Prot:Q9W5Y0,INTERPRO:IPR000718,INTERPRO:IPR008753,INTERPRO:IPR024079,GB_protein:AFH07488,GB_protein:AFH07487,EntrezGene:33005,INTERPRO:IPR042089,DGRC_GENE:FBgn0031081,EPD:Nep3_1,EPD:Nep3_2,EPD:Nep3_3,Fly-FISH:CG9565,FLYGUT:FBgn0031081,GenomeRNAi:33005,iBeetle-Base:TC015207,modMine:FBgn0031081; derived_computed_cyto=19A3-19A3; gbunit=AE014298; MD5=1b695ff9a0ed4e646cc99670d2842f7d; length=8027; release=r6.37; species=Dmel; \n",
      "TAGTTCGAGTGCGATTACCGCTGCTGCTGGTGCTGGAGCTGGCTGTCAGTGGCTCGAGTTGGCAGTGTTGGTGGATACCG\n",
      "TTGCCTTTTTGTGGCTCAGAGCGGTTTGTTTGACCGCGAGCGGAATGCAGATAAGCGCGTTTTAAATGAATCACTGAACG\n",
      "ATCGAATGTTATACCCATCGAGGGTCGGCCAAACTTAATCGAAAGTGCACCAATAAACGGCAAAGGGTCGAGAAAAAGCG\n",
      "AGCGCAGGGAGCTGGAAAAGTTGCATTGCCTTCTTTCTAGCGCACAAAGAAAATGAGCGCAAAGGTAGGCCAAGAGCGCG\n",
      "CCTCTCTTTTGCTGAACGTGTGTGTGTGTGTGTGTGAGTGTGTCAGTGTGTGAGTGTGCTGTATGTGCAATCAGAAGTAC\n",
      "CACCCCACCACACCACACCACTTCCAAACAGTCCGATTCAAAGTCATAATTTCATTTGCATTTGAATAAATGCATCCCAG\n",
      "GCACACCACCCTCTCACCACTCACACATCTAAGCTGCTCCGATTTTATGGTCCTTGGTATAGAGAAATTGTAAGGAAAGC\n",
      "TGCACAAATGAAGTAGCTACCATATGGTATACCAAATAGCCAGGATACTCGCAGCCAAATCGAGAAAGGAGGTGGCATAA\n",
      "CGACAGATAATGACGAGTCGAGAGCACTCATTCTACCAGATTTCAACTCAGTGCAGCTCAAGTGAAATCCCATTGTGGTG\n",
      "\n",
      "==> ../blastdb/dmel-all-predicted-r6.37.fasta <==\n",
      ">:1256737_promoter type=prediction; program=promoter; version=1.0; loc=2L:22282789..22282790; ID=:1256737_promoter; name=6404-6405-AE002603.4-dummy-promoter; MD5=112c61a276d0376f2f6b25dd3a337b1d; length=2; release=r6.37; species=Dmel;\n",
      "CT\n",
      ">:1256738_promoter type=prediction; program=promoter; version=1.0; loc=2L:22289728..22289729; ID=:1256738_promoter; name=13343-13344-AE002603.4-dummy-promoter; MD5=6b712da7b7ccf80851beb06de6c32e6c; length=2; release=r6.37; species=Dmel;\n",
      "AG\n",
      ">:1256739_promoter type=prediction; program=promoter; version=1.0; loc=2L:22295341..22295342; ID=:1256739_promoter; name=18956-18957-AE002603.4-dummy-promoter; MD5=3b98e2dffc6cb06a89dcb0d5c60a0206; length=2; release=r6.37; species=Dmel;\n",
      "AA\n",
      ">:1256740_promoter type=prediction; program=promoter; version=1.0; loc=2L:22298076..22298077; ID=:1256740_promoter; name=21691-21692-AE002603.4-dummy-promoter; MD5=df1f3edb9115acb0a1e04209b7a9937b; length=2; release=r6.37; species=Dmel;\n",
      "TT\n",
      ">:1256741_promoter type=prediction; program=promoter; version=1.0; loc=2L:22306598..22306599; ID=:1256741_promoter; name=30213-30214-AE002603.4-dummy-promoter; MD5=fa868488740aa25870ced6b9169951fb; length=2; release=r6.37; species=Dmel;\n",
      "AT\n"
     ]
    }
   ],
   "source": [
    "!head ../blastdb/d*fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../blastdb/dmel-all-gene-r6.37.fasta\n",
      "../blastdb/dmel-all-predicted-r6.37.fasta\n"
     ]
    }
   ],
   "source": [
    "!ls ../blastdb/d*fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 01/30/2021 08:31:19\n",
      "New DB name:   /Users/sr320/Documents/GitHub/steven-duck/blastdb/dmel-all-gene-r6.37\n",
      "New DB title:  ../blastdb/dmel-all-gene-r6.37.fasta\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 17875 sequences in 1.95157 seconds.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#setting blast program directory\n",
    "bldir = \"/Applications/bioinfo/ncbi-blast-2.11.0+/bin/\"\n",
    "\n",
    "!{bldir}makeblastdb \\\n",
    "-in ../blastdb/dmel-all-gene-r6.37.fasta \\\n",
    "-dbtype nucl \\\n",
    "-out ../blastdb/dmel-all-gene-r6.37"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USAGE\n",
      "  blastn [-h] [-help] [-import_search_strategy filename]\n",
      "    [-export_search_strategy filename] [-task task_name] [-db database_name]\n",
      "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\n",
      "    [-negative_gilist filename] [-negative_seqidlist filename]\n",
      "    [-taxids taxids] [-negative_taxids taxids] [-taxidlist filename]\n",
      "    [-negative_taxidlist filename] [-entrez_query entrez_query]\n",
      "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\n",
      "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\n",
      "    [-out output_file] [-evalue evalue] [-word_size int_value]\n",
      "    [-gapopen open_penalty] [-gapextend extend_penalty]\n",
      "    [-perc_identity float_value] [-qcov_hsp_perc float_value]\n",
      "    [-max_hsps int_value] [-xdrop_ungap float_value] [-xdrop_gap float_value]\n",
      "    [-xdrop_gap_final float_value] [-searchsp int_value]\n",
      "    [-sum_stats bool_value] [-penalty penalty] [-reward reward] [-no_greedy]\n",
      "    [-min_raw_gapped_score int_value] [-template_type type]\n",
      "    [-template_length int_value] [-dust DUST_options]\n",
      "    [-filtering_db filtering_database]\n",
      "    [-window_masker_taxid window_masker_taxid]\n",
      "    [-window_masker_db window_masker_db] [-soft_masking soft_masking]\n",
      "    [-ungapped] [-culling_limit int_value] [-best_hit_overhang float_value]\n",
      "    [-best_hit_score_edge float_value] [-subject_besthit]\n",
      "    [-window_size int_value] [-off_diagonal_range int_value]\n",
      "    [-use_index boolean] [-index_name string] [-lcase_masking]\n",
      "    [-query_loc range] [-strand strand] [-parse_deflines] [-outfmt format]\n",
      "    [-show_gis] [-num_descriptions int_value] [-num_alignments int_value]\n",
      "    [-line_length line_length] [-html] [-sorthits sort_hits]\n",
      "    [-sorthsps sort_hsps] [-max_target_seqs num_sequences]\n",
      "    [-num_threads int_value] [-remote] [-version]\n",
      "\n",
      "DESCRIPTION\n",
      "   Nucleotide-Nucleotide BLAST 2.11.0+\n",
      "\n",
      "OPTIONAL ARGUMENTS\n",
      " -h\n",
      "   Print USAGE and DESCRIPTION;  ignore all other parameters\n",
      " -help\n",
      "   Print USAGE, DESCRIPTION and ARGUMENTS; ignore all other parameters\n",
      " -version\n",
      "   Print version number;  ignore other arguments\n",
      "\n",
      " *** Input query options\n",
      " -query <File_In>\n",
      "   Input file name\n",
      "   Default = `-'\n",
      " -query_loc <String>\n",
      "   Location on the query sequence in 1-based offsets (Format: start-stop)\n",
      " -strand <String, `both', `minus', `plus'>\n",
      "   Query strand(s) to search against database/subject\n",
      "   Default = `both'\n",
      "\n",
      " *** General search options\n",
      " -task <String, Permissible values: 'blastn' 'blastn-short' 'dc-megablast'\n",
      "                'megablast' 'rmblastn' >\n",
      "   Task to execute\n",
      "   Default = `megablast'\n",
      " -db <String>\n",
      "   BLAST database name\n",
      "    * Incompatible with:  subject, subject_loc\n",
      " -out <File_Out>\n",
      "   Output file name\n",
      "   Default = `-'\n",
      " -evalue <Real>\n",
      "   Expectation value (E) threshold for saving hits \n",
      "   Default = `10'\n",
      " -word_size <Integer, >=4>\n",
      "   Word size for wordfinder algorithm (length of best perfect match)\n",
      " -gapopen <Integer>\n",
      "   Cost to open a gap\n",
      " -gapextend <Integer>\n",
      "   Cost to extend a gap\n",
      " -penalty <Integer, <=0>\n",
      "   Penalty for a nucleotide mismatch\n",
      " -reward <Integer, >=0>\n",
      "   Reward for a nucleotide match\n",
      " -use_index <Boolean>\n",
      "   Use MegaBLAST database index\n",
      "   Default = `false'\n",
      " -index_name <String>\n",
      "   MegaBLAST database index name (deprecated; use only for old style indices)\n",
      "\n",
      " *** BLAST-2-Sequences options\n",
      " -subject <File_In>\n",
      "   Subject sequence(s) to search\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\n",
      "   negative_seqidlist, taxids, taxidlist, negative_taxids, negative_taxidlist,\n",
      "   db_soft_mask, db_hard_mask\n",
      " -subject_loc <String>\n",
      "   Location on the subject sequence in 1-based offsets (Format: start-stop)\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\n",
      "   negative_seqidlist, taxids, taxidlist, negative_taxids, negative_taxidlist,\n",
      "   db_soft_mask, db_hard_mask, remote\n",
      "\n",
      " *** Formatting options\n",
      " -outfmt <String>\n",
      "   alignment view options:\n",
      "     0 = Pairwise,\n",
      "     1 = Query-anchored showing identities,\n",
      "     2 = Query-anchored no identities,\n",
      "     3 = Flat query-anchored showing identities,\n",
      "     4 = Flat query-anchored no identities,\n",
      "     5 = BLAST XML,\n",
      "     6 = Tabular,\n",
      "     7 = Tabular with comment lines,\n",
      "     8 = Seqalign (Text ASN.1),\n",
      "     9 = Seqalign (Binary ASN.1),\n",
      "    10 = Comma-separated values,\n",
      "    11 = BLAST archive (ASN.1),\n",
      "    12 = Seqalign (JSON),\n",
      "    13 = Multiple-file BLAST JSON,\n",
      "    14 = Multiple-file BLAST XML2,\n",
      "    15 = Single-file BLAST JSON,\n",
      "    16 = Single-file BLAST XML2,\n",
      "    17 = Sequence Alignment/Map (SAM),\n",
      "    18 = Organism Report\n",
      "   \n",
      "   Options 6, 7, 10 and 17 can be additionally configured to produce\n",
      "   a custom format specified by space delimited format specifiers,\n",
      "   or in the case of options 6, 7, and 10, by a token specified\n",
      "   by the delim keyword. E.g.: \"17 delim=@ qacc sacc score\".\n",
      "   The delim keyword must appear after the numeric output format\n",
      "   specification.\n",
      "   The supported format specifiers for options 6, 7 and 10 are:\n",
      "   \t    qseqid means Query Seq-id\n",
      "   \t       qgi means Query GI\n",
      "   \t      qacc means Query accesion\n",
      "   \t   qaccver means Query accesion.version\n",
      "   \t      qlen means Query sequence length\n",
      "   \t    sseqid means Subject Seq-id\n",
      "   \t sallseqid means All subject Seq-id(s), separated by a ';'\n",
      "   \t       sgi means Subject GI\n",
      "   \t    sallgi means All subject GIs\n",
      "   \t      sacc means Subject accession\n",
      "   \t   saccver means Subject accession.version\n",
      "   \t   sallacc means All subject accessions\n",
      "   \t      slen means Subject sequence length\n",
      "   \t    qstart means Start of alignment in query\n",
      "   \t      qend means End of alignment in query\n",
      "   \t    sstart means Start of alignment in subject\n",
      "   \t      send means End of alignment in subject\n",
      "   \t      qseq means Aligned part of query sequence\n",
      "   \t      sseq means Aligned part of subject sequence\n",
      "   \t    evalue means Expect value\n",
      "   \t  bitscore means Bit score\n",
      "   \t     score means Raw score\n",
      "   \t    length means Alignment length\n",
      "   \t    pident means Percentage of identical matches\n",
      "   \t    nident means Number of identical matches\n",
      "   \t  mismatch means Number of mismatches\n",
      "   \t  positive means Number of positive-scoring matches\n",
      "   \t   gapopen means Number of gap openings\n",
      "   \t      gaps means Total number of gaps\n",
      "   \t      ppos means Percentage of positive-scoring matches\n",
      "   \t    frames means Query and subject frames separated by a '/'\n",
      "   \t    qframe means Query frame\n",
      "   \t    sframe means Subject frame\n",
      "   \t      btop means Blast traceback operations (BTOP)\n",
      "   \t    staxid means Subject Taxonomy ID\n",
      "   \t  ssciname means Subject Scientific Name\n",
      "   \t  scomname means Subject Common Name\n",
      "   \tsblastname means Subject Blast Name\n",
      "   \t sskingdom means Subject Super Kingdom\n",
      "   \t   staxids means unique Subject Taxonomy ID(s), separated by a ';'\n",
      "   \t\t\t (in numerical order)\n",
      "   \t sscinames means unique Subject Scientific Name(s), separated by a ';'\n",
      "   \t scomnames means unique Subject Common Name(s), separated by a ';'\n",
      "   \tsblastnames means unique Subject Blast Name(s), separated by a ';'\n",
      "   \t\t\t (in alphabetical order)\n",
      "   \tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\n",
      "   \t\t\t (in alphabetical order) \n",
      "   \t    stitle means Subject Title\n",
      "   \tsalltitles means All Subject Title(s), separated by a '<>'\n",
      "   \t   sstrand means Subject Strand\n",
      "   \t     qcovs means Query Coverage Per Subject\n",
      "   \t   qcovhsp means Query Coverage Per HSP\n",
      "   \t    qcovus means Query Coverage Per Unique Subject (blastn only)\n",
      "   When not provided, the default value is:\n",
      "   'qaccver saccver pident length mismatch gapopen qstart qend sstart send\n",
      "   evalue bitscore', which is equivalent to the keyword 'std'\n",
      "   The supported format specifier for option 17 is:\n",
      "   \t        SQ means Include Sequence Data\n",
      "   \t        SR means Subject as Reference Seq\n",
      "   Default = `0'\n",
      " -show_gis\n",
      "   Show NCBI GIs in deflines?\n",
      " -num_descriptions <Integer, >=0>\n",
      "   Number of database sequences to show one-line descriptions for\n",
      "   Not applicable for outfmt > 4\n",
      "   Default = `500'\n",
      "    * Incompatible with:  max_target_seqs\n",
      " -num_alignments <Integer, >=0>\n",
      "   Number of database sequences to show alignments for\n",
      "   Default = `250'\n",
      "    * Incompatible with:  max_target_seqs\n",
      " -line_length <Integer, >=1>\n",
      "   Line length for formatting alignments\n",
      "   Not applicable for outfmt > 4\n",
      "   Default = `60'\n",
      " -html\n",
      "   Produce HTML output?\n",
      " -sorthits <Integer, (>=0 and =<4)>\n",
      "   Sorting option for hits:\n",
      "   alignment view options:\n",
      "     0 = Sort by evalue,\n",
      "     1 = Sort by bit score,\n",
      "     2 = Sort by total score,\n",
      "     3 = Sort by percent identity,\n",
      "     4 = Sort by query coverage\n",
      "   Not applicable for outfmt > 4\n",
      " -sorthsps <Integer, (>=0 and =<4)>\n",
      "   Sorting option for hps:\n",
      "     0 = Sort by hsp evalue,\n",
      "     1 = Sort by hsp score,\n",
      "     2 = Sort by hsp query start,\n",
      "     3 = Sort by hsp percent identity,\n",
      "     4 = Sort by hsp subject start\n",
      "   Not applicable for outfmt != 0\n",
      "\n",
      " *** Query filtering options\n",
      " -dust <String>\n",
      "   Filter query sequence with DUST (Format: 'yes', 'level window linker', or\n",
      "   'no' to disable)\n",
      "   Default = `20 64 1'\n",
      " -filtering_db <String>\n",
      "   BLAST database containing filtering elements (i.e.: repeats)\n",
      " -window_masker_taxid <Integer>\n",
      "   Enable WindowMasker filtering using a Taxonomic ID\n",
      " -window_masker_db <String>\n",
      "   Enable WindowMasker filtering using this repeats database.\n",
      " -soft_masking <Boolean>\n",
      "   Apply filtering locations as soft masks\n",
      "   Default = `true'\n",
      " -lcase_masking\n",
      "   Use lower case filtering in query and subject sequence(s)?\n",
      "\n",
      " *** Restrict search or results\n",
      " -gilist <String>\n",
      "   Restrict search of database to list of GIs\n",
      "    * Incompatible with:  seqidlist, taxids, taxidlist, negative_gilist,\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
      "   subject_loc\n",
      " -seqidlist <String>\n",
      "   Restrict search of database to list of SeqIDs\n",
      "    * Incompatible with:  gilist, taxids, taxidlist, negative_gilist,\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
      "   subject_loc\n",
      " -negative_gilist <String>\n",
      "   Restrict search of database to everything except the specified GIs\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
      "   subject_loc\n",
      " -negative_seqidlist <String>\n",
      "   Restrict search of database to everything except the specified SeqIDs\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
      "   negative_gilist, negative_taxids, negative_taxidlist, remote, subject,\n",
      "   subject_loc\n",
      " -taxids <String>\n",
      "   Restrict search of database to include only the specified taxonomy IDs\n",
      "   (multiple IDs delimited by ',')\n",
      "    * Incompatible with:  gilist, seqidlist, taxidlist, negative_gilist,\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
      "   subject_loc\n",
      " -negative_taxids <String>\n",
      "   Restrict search of database to everything except the specified taxonomy IDs\n",
      "   (multiple IDs delimited by ',')\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
      "   negative_gilist, negative_seqidlist, negative_taxidlist, remote, subject,\n",
      "   subject_loc\n",
      " -taxidlist <String>\n",
      "   Restrict search of database to include only the specified taxonomy IDs\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, negative_gilist,\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\n",
      "   subject_loc\n",
      " -negative_taxidlist <String>\n",
      "   Restrict search of database to everything except the specified taxonomy IDs\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
      "   negative_gilist, negative_seqidlist, negative_taxids, remote, subject,\n",
      "   subject_loc\n",
      " -entrez_query <String>\n",
      "   Restrict search with the given Entrez query\n",
      "    * Requires:  remote\n",
      " -db_soft_mask <String>\n",
      "   Filtering algorithm ID to apply to the BLAST database as soft masking\n",
      "    * Incompatible with:  db_hard_mask, subject, subject_loc\n",
      " -db_hard_mask <String>\n",
      "   Filtering algorithm ID to apply to the BLAST database as hard masking\n",
      "    * Incompatible with:  db_soft_mask, subject, subject_loc\n",
      " -perc_identity <Real, 0..100>\n",
      "   Percent identity\n",
      " -qcov_hsp_perc <Real, 0..100>\n",
      "   Percent query coverage per hsp\n",
      " -max_hsps <Integer, >=1>\n",
      "   Set maximum number of HSPs per subject sequence to save for each query\n",
      " -culling_limit <Integer, >=0>\n",
      "   If the query range of a hit is enveloped by that of at least this many\n",
      "   higher-scoring hits, delete the hit\n",
      "    * Incompatible with:  best_hit_overhang, best_hit_score_edge\n",
      " -best_hit_overhang <Real, (>0 and <0.5)>\n",
      "   Best Hit algorithm overhang value (recommended value: 0.1)\n",
      "    * Incompatible with:  culling_limit\n",
      " -best_hit_score_edge <Real, (>0 and <0.5)>\n",
      "   Best Hit algorithm score edge value (recommended value: 0.1)\n",
      "    * Incompatible with:  culling_limit\n",
      " -subject_besthit\n",
      "   Turn on best hit per subject sequence\n",
      " -max_target_seqs <Integer, >=1>\n",
      "   Maximum number of aligned sequences to keep \n",
      "   (value of 5 or more is recommended)\n",
      "   Default = `500'\n",
      "    * Incompatible with:  num_descriptions, num_alignments\n",
      "\n",
      " *** Discontiguous MegaBLAST options\n",
      " -template_type <String, `coding', `coding_and_optimal', `optimal'>\n",
      "   Discontiguous MegaBLAST template type\n",
      "    * Requires:  template_length\n",
      " -template_length <Integer, Permissible values: '16' '18' '21' >\n",
      "   Discontiguous MegaBLAST template length\n",
      "    * Requires:  template_type\n",
      "\n",
      " *** Statistical options\n",
      " -dbsize <Int8>\n",
      "   Effective length of the database \n",
      " -searchsp <Int8, >=0>\n",
      "   Effective length of the search space\n",
      " -sum_stats <Boolean>\n",
      "   Use sum statistics\n",
      "\n",
      " *** Search strategy options\n",
      " -import_search_strategy <File_In>\n",
      "   Search strategy to use\n",
      "    * Incompatible with:  export_search_strategy\n",
      " -export_search_strategy <File_Out>\n",
      "   File name to record the search strategy used\n",
      "    * Incompatible with:  import_search_strategy\n",
      "\n",
      " *** Extension options\n",
      " -xdrop_ungap <Real>\n",
      "   X-dropoff value (in bits) for ungapped extensions\n",
      " -xdrop_gap <Real>\n",
      "   X-dropoff value (in bits) for preliminary gapped extensions\n",
      " -xdrop_gap_final <Real>\n",
      "   X-dropoff value (in bits) for final gapped alignment\n",
      " -no_greedy\n",
      "   Use non-greedy dynamic programming extension\n",
      " -min_raw_gapped_score <Integer>\n",
      "   Minimum raw gapped score to keep an alignment in the preliminary gapped and\n",
      "   traceback stages\n",
      " -ungapped\n",
      "   Perform ungapped alignment only?\n",
      " -window_size <Integer, >=0>\n",
      "   Multiple hits window size, use 0 to specify 1-hit algorithm\n",
      " -off_diagonal_range <Integer, >=0>\n",
      "   Number of off-diagonals to search for the 2nd hit, use 0 to turn off\n",
      "   Default = `0'\n",
      "\n",
      " *** Miscellaneous options\n",
      " -parse_deflines\n",
      "   Should the query and subject defline(s) be parsed?\n",
      " -num_threads <Integer, >=1>\n",
      "   Number of threads (CPUs) to use in the BLAST search\n",
      "   Default = `1'\n",
      "    * Incompatible with:  remote\n",
      " -remote\n",
      "   Execute search remotely?\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\n",
      "   negative_gilist, negative_seqidlist, negative_taxids, negative_taxidlist,\n",
      "   subject_loc, num_threads\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!{bldir}blastn -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: [blastn] Examining 5 or more matches is recommended\n"
     ]
    }
   ],
   "source": [
    "!{bldir}blastn \\\n",
    "-task blastn \\\n",
    "-query ../data/GCF_000297895.1_oyster_v9_cds_from_genomic.fna \\\n",
    "-db ../blastdb/dmel-all-gene-r6.37  \\\n",
    "-out ../analyses/Cg-dmel-gene_blastn.tab \\\n",
    "-evalue 1E-20 \\\n",
    "-num_threads 2 \\\n",
    "-max_target_seqs 1 \\\n",
    "-max_hsps 1 \\\n",
    "-outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "lcl|NW_011934528.1_cds_XP_019926840.1_21\tFBgn0031497\t72.381\t945\t253\t8\t442\t1382\t698\t1638\t9.27e-140\t499\n",
      "lcl|NW_011934533.1_cds_XP_011449976.1_33\tFBgn0000559\t71.355\t974\t263\t10\t1243\t2208\t2763\t3728\t2.41e-130\t469\n",
      "lcl|NW_011934555.1_cds_XP_011425356.1_65\tFBgn0000667\t77.262\t409\t93\t0\t95\t503\t11451\t11859\t3.15e-85\t319\n",
      "lcl|NW_011934555.1_cds_XP_011425276.1_66\tFBgn0000667\t77.262\t409\t93\t0\t95\t503\t11451\t11859\t3.15e-85\t319\n",
      "lcl|NW_011934557.1_cds_XP_011425759.1_69\tFBgn0051155\t71.010\t307\t89\t0\t14\t320\t159\t465\t5.10e-36\t153\n",
      "lcl|NW_011934557.1_cds_NP_001292274.1_71\tFBgn0038834\t81.395\t129\t24\t0\t205\t333\t642\t770\t5.37e-28\t125\n",
      "lcl|NW_011934558.1_cds_XP_011426919.1_83\tFBgn0027329\t67.036\t1265\t407\t5\t138\t1394\t661\t1923\t1.26e-106\t389\n",
      "lcl|NW_011934558.1_cds_XP_011427533.1_88\tFBgn0026576\t70.325\t246\t73\t0\t682\t927\t2093\t2338\t3.02e-24\t115\n",
      "lcl|NW_011934558.1_cds_XP_011427448.1_89\tFBgn0026576\t70.325\t246\t73\t0\t751\t996\t2093\t2338\t3.20e-24\t115\n",
      "lcl|NW_011934575.1_cds_XP_011433497.1_137\tFBgn0015795\t73.023\t215\t58\t0\t178\t392\t1236\t1450\t2.47e-28\t127\n"
     ]
    }
   ],
   "source": [
    "!head ../analyses/Cg-dmel-gene_blastn.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#setting blast program directory\n",
    "bldir = \"/Applications/bioinfo/ncbi-blast-2.11.0+/bin/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: [blastn] Examining 5 or more matches is recommended\n"
     ]
    }
   ],
   "source": [
    "!{bldir}blastn \\\n",
    "-task blastn \\\n",
    "-query ../data/GCF_000297895.1_oyster_v9_cds_from_genomic.fna \\\n",
    "-db ../blastdb/dmel-all-gene-r6.37  \\\n",
    "-out ../analyses/Cg-dmel-gene_blastn_02.tab \\\n",
    "-evalue 1E-05 \\\n",
    "-num_threads 1 \\\n",
    "-max_target_seqs 1 \\\n",
    "-max_hsps 1 \\\n",
    "-outfmt \"6 qaccver saccver evalue\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "lcl|NW_011934527.1_cds_XP_011437695.1_20\tFBgn0037660\t1.30e-13\n",
      "lcl|NW_011934528.1_cds_XP_019926840.1_21\tFBgn0031497\t9.27e-140\n",
      "lcl|NW_011934533.1_cds_XP_011449976.1_33\tFBgn0000559\t2.41e-130\n",
      "lcl|NW_011934534.1_cds_XP_011452277.1_35\tFBgn0032799\t1.08e-14\n",
      "lcl|NW_011934534.1_cds_XP_011453429.1_36\tFBgn0263110\t4.87e-07\n",
      "lcl|NW_011934535.1_cds_XP_019918435.1_37\tFBgn0001197\t6.58e-07\n",
      "lcl|NW_011934548.1_cds_XP_011424376.1_54\tFBgn0038499\t1.24e-06\n",
      "lcl|NW_011934548.1_cds_XP_019922042.1_55\tFBgn0038499\t1.23e-06\n",
      "lcl|NW_011934548.1_cds_XP_011424190.1_56\tFBgn0262871\t8.50e-20\n",
      "lcl|NW_011934555.1_cds_XP_011425356.1_65\tFBgn0000667\t3.15e-85\n"
     ]
    }
   ],
   "source": [
    "!head ../analyses/Cg-dmel-gene_blastn_02.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
